eogrow/utils/map.py

```diff
@@ -163,7 +168,22 @@ def merge_tiffs(
     if dtype is not None:
         gdalwarp_options += f" -ot {GDAL_DTYPE_SETTINGS[dtype]}"
 
-    command = f"gdalwarp {gdalwarp_options} {' '.join(input_filenames)} {merged_filename}"
+    input_filelist = list(input_filenames)
+    command = f"gdalwarp {gdalwarp_options} {' '.join(input_filelist)} {merged_filename}"
+
+    if len(command) > SH_COMMAND_LIMIT or len(input_filelist) > OPEN_FILES_LIMIT:
+        # generate a text file with
+        merged_path = Path(merged_filename)
+        tile_list_path = str(merged_path.with_name(f"{merged_path.stem}_files.txt"))
+        vrt_file_path = str(merged_path.with_name(f"{merged_path.stem}_temp.vrt"))
+        with open(tile_list_path, "w") as file:
+            for string in input_filelist:
```

Comment:
Probably a personal preference, but this could be rewritten as 
```
with open(vrt_file_path, "w") as vrt_file:
    vrt_file.writelines([f"{file}\n" for file in input_filelist])
```

Reply:
resolved